### PR TITLE
LSR-1036 iOS and Android changes for Xamarin binding

### PR DESCRIPTION
**Pull request by dgarciasarai**
*iOS*:
- Add name to classes annotated with @objc: It's necessary to add an understandable name to classes annotated with @objc to generate the Xamarin iOS bindings and to use understandable names in Xamarin.iOS solutions.
- Remove module name in default views.
- SF

*Android*:
- All screenlets should have getter and setter for their listener property: Xamarin binding will generate a property called `Listener` and with that we can subscribe to listener events.

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java [hunk 1]
```diff
@@ -278,18 +278,34 @@ public void setPortletItemName(String portletItemName) {
...
+	public AssetDisplayListener getListener() {
+		return listener;
+	}
+
 	public void setListener(AssetDisplayListener listener) {
 		this.listener = listener;
 	}
 
+	public AssetDisplayInnerScreenletListener getConfigureListener() {
+		return configureListener;
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: configuration? other name?

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java [hunk 2]
```diff
@@ -278,18 +278,34 @@ public void setPortletItemName(String portletItemName) {
...
 
+	public boolean isAutoLoad() {
+		return autoLoad;
+	}
+
 	public void setAutoLoad(boolean autoLoad) {
 		this.autoLoad = autoLoad;
 	}
 
+	public HashMap<String, Integer> getLayouts() {
```
Comment (nhpatt) [inline android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: Expose Map if you can

File: ios/Framework/Core/Context/Storage/BaseCredentialsStoreKeyChain.swift [hunk 3]
```diff
@@ -78,14 +80,41 @@ import UIKit
...
 	}
 
-	open func loadStoredCredentials() -> Bool {
+	open dynamic func loadStoredCredentials() -> Bool {
 		return loadStoredCredentials(shouldLoadServer: false)
 	}
 
-	open func loadStoredCredentialsAndServer() -> Bool {
+	open dynamic func loadStoredCredentialsAndServer() -> Bool {
 		return loadStoredCredentials(shouldLoadServer: true)
```
Comment (victorg1991) [inline ios/Framework/Core/Context/Storage/BaseCredentialsStoreKeyChain.swift]: Do we need to be able to override these methods from xamarin?

File: ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m [hunk 4]
```diff
@@ -20,7 +20,7 @@ @implementation ScreensBridgePlugin
 
 - (void)postMessage:(CDVInvokedUrlCommand*)command {
 
-	CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@""];
+	/*CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@""];
```
Comment (victorg1991) [inline ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m]: Cordova won't work if this is commented

File: ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m [hunk 5]
```diff
@@ -31,7 +31,7 @@ - (void)postMessage:(CDVInvokedUrlCommand*)command {
 		[vc handleJsCallWithNamespace:namespace message: message];
 	});
 
-	[self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
+	[self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];*/
```
Comment (victorg1991) [inline ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m]: same

File: ios/Framework/LiferayScreens.xcodeproj/xcshareddata/xcschemes/LiferayScreens.xcscheme [hunk 6]
```diff
@@ -66,9 +67,10 @@
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
```
Comment (victorg1991) [inline ios/Framework/LiferayScreens.xcodeproj/xcshareddata/xcschemes/LiferayScreens.xcscheme]: What do you think about creating a new target for building the framework needed for Xamarin? (including disable bitcode) This way we don't need to worry about setting the build configuration to debug again (which compiles faster) :)

**Comment by victorg1991** [comment]
LGTM with some nits 👏🏻, can you also fix the format issues? Running swiftlint autocorrect will solve most of them, if not all

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java (hunk 1 above)
Comment (dgarciasarai) [reply android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: The listener was created to configure the "child" (inner) screenlet or render a custom view. So, can we call it... innerScreenletListener? configurationListener?

File: android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java (hunk 2 above)
Comment (dgarciasarai) [reply android/library/src/main/java/com/liferay/mobile/screens/asset/display/AssetDisplayScreenlet.java]: Map instead HashMap?

File: android/samples/test-app/build.gradle [hunk 7]
```diff
@@ -21,6 +21,7 @@ android {
 	}
 	lintOptions {
 		abortOnError false
+    disable 'MissingTranslation'
```
[the remark this answers is not in the document]
Comment (dgarciasarai) [reply android/samples/test-app/build.gradle]: When you are going to create the `.jar` file, gradle complains that some translations are missing :(

File: android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/MainActivity.java [hunk 8]
```diff
@@ -166,7 +166,7 @@ public void onClick(View v) {
 				break;
 			case R.id.user_display:
 				Intent intentUser = getIntentWithTheme(AssetDisplayActivity.class);
-				intentUser.putExtra("entryId", Long.valueOf(getResources().getString(R.string.liferay_user_entryId)));
+				intentUser.putExtra("entryId", Long.valueOf(getResources().getString(R.string.liferay_video_entryId)));
 				DefaultAnimation.startActivityWithAnimation(this, intentUser);
```
[the remark this answers is not in the document]
Comment (dgarciasarai) [reply android/samples/test-app/src/main/java/com/liferay/mobile/screens/testapp/MainActivity.java]: Upsss...

File: ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m (hunk 4 above)
Comment (dgarciasarai) [reply ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m]: sorry, this was commented to do some tests :)

File: ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m (hunk 5 above)
Comment (dgarciasarai) [reply ios/Framework/Themes/Default/Portlet/cordova/ScreensBridgePlugin.m]: same as before, sorry

File: ios/Framework/LiferayScreens.xcodeproj/xcshareddata/xcschemes/LiferayScreens.xcscheme (hunk 6 above)
Comment (dgarciasarai) [reply ios/Framework/LiferayScreens.xcodeproj/xcshareddata/xcschemes/LiferayScreens.xcscheme]: Great idea!

File: ios/Framework/Core/Context/Storage/BaseCredentialsStoreKeyChain.swift (hunk 3 above)
Comment (dgarciasarai) [reply ios/Framework/Core/Context/Storage/BaseCredentialsStoreKeyChain.swift]: You can call super and then store the credentials somewhere else in Xamarin, I don't know 🤷‍♀️  why @victorg1991 ?

Comment (victorg1991) [reply ios/Framework/Core/Context/Storage/BaseCredentialsStoreKeyChain.swift]: I didn't see this message, I meant that this class won't be override in xamarin, right? You only need to override this class if you want to create a new CredentialsStore

**Pull request merged**